litrpc/netcmds.go

```diff
@@ -64,6 +65,29 @@ func (r *LitRPC) Connect(args ConnectArgs, reply *StatusReply) error {
 	return nil
 }
 
+// ------------------------- name a connection
+type AssignNicknameArgs struct {
+	Peer     uint32
+	Nickname string
+}
+
+func (r *LitRPC) AssignNickname(args AssignNicknameArgs, reply *StatusReply) error {
+	err := r.Node.SaveNicknameForPeerIdx(args.Nickname, args.Peer)
+	if err != nil {
+		return err
+	}
+
+	peer, ok := r.Node.RemoteCons[args.Peer]
+	if !ok {
+		return fmt.Errorf("peer %d doesn't exist", args.Peer)
```

Comment:
You might be on to something though, I was able to crash my node by sending a request a call to AssignNickname, I didn't note where the error was, I'll try again.